pkg/tasks/controlplane.go

```diff
@@ -55,6 +56,23 @@ func (t *ControlPlaneTask) Run(ctx context.Context) error {
 		return errors.Wrap(err, "reconciling control-plane kubelet ServiceMonitor failed")
 	}
 
+	smkpa, err := t.factory.ControlPlaneKubeletServiceMonitorPA()
+	if err != nil {
+		return errors.Wrap(err, "initializing prometheus-adapater dedicated control-plane kubelet ServiceMonitor failed")
+	}
+
+	if *t.config.ClusterMonitoringConfiguration.K8sPrometheusAdapter.DedicatedServiceMonitors.Enabled {
+		err = t.client.CreateOrUpdateServiceMonitor(ctx, smkpa)
+		if err != nil {
+			return errors.Wrap(err, "reconcilin prometheus-adapater dedicatedg control-plane kubelet ServiceMonitor failed")
+		}
+	} else {
+		err = t.client.DeleteServiceMonitor(ctx, smkpa)
+		if err != nil {
+			return errors.Wrap(err, "deleting prometheus-adapater dedicated control-plane etcd ServiceMonitor failed")
```

Comment:
I'm not 100% sure, but aren't we going to land here every time the DedicatedServiceMonitors are not enabled? Shouldn't we exclude the IfNotFound error?

Reply:
Yes and we do: https://github.com/openshift/cluster-monitoring-operator/blob/ba2f4472bc07989f459cbf5f299bac8100d193df/pkg/client/client.go#L742-L756

Reply:
Ahhh the check was there, disregard then